packages/create-cloudflare/src/helpers/args.ts

```diff
@@ -1,88 +1,239 @@
+  `,
+	positionals: [
+		{
+			name: "directory",
 			type: "string",
-			description:
-				"The directory where the application should be created. The name of the application is taken from the directory name",
-		})
-		.option("type", {
+			description: `The directory where the application should be created. Also used as the name of the application.
+
+        If a path is provided that includes intermediary directories, only the base name will be used as the name of the application.`,
+		},
+	],
+	options: [
+		{
+			name: "type",
+			alias: "t",
 			type: "string",
-			requiresArg: true,
-			description: "The type of application that should be created",
-		})
-		.option("framework", {
+			description: `
+        When using a built-in template, specifies the type of application that should be created.
+
+        Note that "--type" and "--template" are mutually exclusive options. If both are provided, "--type" will be used.
+        `,
+			values: [
+				{
+					name: "web-framework",
+					description: "A website or web application.",
+				},
+				{
+					name: "hello-world",
+					description: "A basic “Hello World” Cloudflare Worker.",
+				},
+				{
+					name: "hello-world-durable-object",
+					description:
+						"A basic “Hello World” Cloudflare Worker with a Durable Worker.",
+				},
+				{
+					name: "common",
+					description:
+						"A Cloudflare Worker which implements a common example of routing/proxying functionalities.",
+				},
+				{
+					name: "scheduled",
+					description:
+						"A scheduled Cloudflare Worker (triggered via Cron Triggers).",
+				},
+				{
+					name: "queues",
+					description:
+						"A Cloudflare Worker which is both a consumer and produced of Queues.",
+				},
+				{
+					name: "openapi",
+					description: "A Worker implementing an OpenAPI REST endpoint.",
+				},
+			],
+		},
+		{
+			name: "framework",
+			alias: "f",
 			type: "string",
-			requiresArg: true,
-			description:
-				"The type of framework to use to create a web application (when using this option `--type` is ignored)",
-		})
-		.option("deploy", {
+			description: `The type of framework to use to create a web application (when using this option "--type" is coerced to "web-framework")
+
+      When using the --framework option, C3 will dispatch to the official creation tool used by the framework (ex. "create-remix" is used for Remix).
+
+      You may specify additional arguments to be passed directly to these underlying tools by adding them after "--" in the argument string, like so:
+
+      npm create cloudflare -- --framework next -- --ts
+
+      Note that there are 2 occurrences of "--" in the above arg string. The first "--" is required by "npm create" when passing --flags. The second "--" signals to C3 that the remaining arguments should be passed to "create-next-app".
```

Comment:
instead of this and having to clarify the double `--`s I think you could say something like:

```
You may specify additional arguments to be passed directly to these underlying tools by adding them after a "--" argument, like so:
       npm create cloudflare -- --framework next -- --ts
       pnpm create clouldfare --framework next -- --ts
```

so providing multiple PMs here could clarify the `--` usage?

I'm not really sure, it's just a thought 🙂